server/src/main/java/org/elasticsearch/script/ScriptContextStats.java

```diff
@@ -43,6 +59,68 @@ public void writeTo(StreamOutput out) throws IOException {
         out.writeVLong(compilations);
         out.writeVLong(cacheEvictions);
         out.writeVLong(compilationLimitTriggered);
+        if (out.getVersion().onOrAfter(Version.V_8_0_0)) {
+            out.writeList(List.of(compilationsHistory, cacheEvictionsHistory));
+        }
+    }
+
+    public static class TimeSeries implements Writeable, ToXContentFragment {
+        public final long five;
+        public final long fifteen;
+        public final long day;
+
+        public TimeSeries() {
+            this.five = 0;
+            this.fifteen = 0;
+            this.day = 0;
+        }
+
+        public TimeSeries(long five, long fifteen, long day) {
+            assert five >= 0;
+            this.five = five;
+            assert fifteen >= five;
+            this.fifteen = fifteen;
+            assert day >= fifteen;
+            this.day = day;
+        }
+
+        public TimeSeries(StreamInput in) throws IOException {
+            five = in.readVLong();
+            fifteen = in.readVLong();
+            day = in.readVLong();
+        }
+
+        @Override
+        public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
+            builder.field(Fields.FIVE_MINUTES, five);
+            builder.field(Fields.FIFTEEN_MINUTES, fifteen);
+            builder.field(Fields.TWENTY_FOUR_HOURS, day);
```

Comment:
Maybe have this constant be Fields.ONE_DAY to match the variable name or the variable name be twentyFourHours.

Reply:
Done.